bikes/views.py

```diff
@@ -28,6 +34,47 @@ def create(self, request, *args, **kwargs):
         )
 
 
-class RentedBikesListAPIView(ListAPIView):
+class RentedBikesViewSet(CreateModelMixin, ListModelMixin, viewsets.GenericViewSet):
     queryset = Bike.objects.filter(status=BikeStatus.rented)
-    serializer_class = ReadBikeSerializer
+
+    def get_serializer_class(self):
+        if self.action == "create":
+            return RentBikeSerializer
+        else:
+            return ReadBikeSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        headers = self.get_success_headers(serializer.data)
+        try:
+            rented_bike = Bike.objects.get(id=request.data.get("id"))
+        except Bike.DoesNotExist:
+            return Response(
+                {"message": "bike not found"}, status=status.HTTP_404_NOT_FOUND
+            )
+        if rented_bike.user.username == "blocked":  # placeholder user is blocked
```

Comment:
Pisanie kodu który nie ma prawa działać na zaś to mocny antipattern, nie rób tak nigdy proszę.
Zastąp to zakomentowanym blokiem kodu z obowiązkowym TODO (takie komentarze łatwo wyszukać potem w całym projekcie.

Moja propozycja
```
        # TODO(kkrolik): uncomment once blocking users is introduced
        # user = somehow_get_user_from_token_in_headers(request)
        # if user.status == UserStatus.blocked:
        #     return Response(
        #         {"message": "Blocked users are not allowed to rent bikes"},
        #         status=status.HTTP_403_FORBIDDEN,
        #     )
```